### DISCUSSION TITLE
Issue with PyAnsys Installation: ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by t0xine** [ansys/pyansys#569] [Q&A] [answered]
Dear all,

I hope this message finds you well. I am reaching out to seek assistance with an issue I encountered while attempting to use PyAnsys.

I'm working with Python on VS code.

Upon installing PyAnsys via pip and attempting to run a script, I encountered the following error message:

ModuleNotFoundError: No module named 'ansys.geometry'; 'ansys' is not a package

I have already taken the following steps to troubleshoot the issue:

     * Checked the installation of PyAnsys to ensure it was successful.
     * Ensured that all dependencies required by PyAnsys are installed.
     * Confirmed that I am using the correct Python interpreter and environment.
     * Attempted to isolate the issue by running the script in a virtual environment.

Despite these efforts, I am still unable to resolve the error. I would greatly appreciate any guidance or assistance you can provide to help me resolve this issue and successfully utilize PyAnsys for my project.

The package ansys is in the same directory as numpy:

C:\Users\x\AppData\Local\Programs\Python\Python312\Lib\site-packages\ansys\geometry\core

C:\Users\x\AppData\Local\Programs\Python\Python312\Lib\site-packages\numpy

VS code has no problem running numpy but cannot run ansys.

import ansys
print(dir(ansys))

output:

['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'ansys']

If you require any additional information or logs to assist with troubleshooting, please don't hesitate to ask. Your help in resolving this matter would be greatly appreciated.

Thank you very much for your time and support.

Best regards,

Julien

**Comment by RobPasMue** [comment]
Hi @t0xine!

Thanks for reaching out, let me try to support you in your request. Please, verify with me the following:

1. Is the ``pyansys`` package installed?
If so, running the following command on PowerShell/CMD should give you a similar output to mine:
```bash
>>> pip freeze
...
pyansys==???????
...
```
If the above is not true, please run ``pip install pyansys``

2. Which version of ``pyansys`` is installed?
This would be the ``???????`` in the previous output.

3. What is the output of running the following code snippet in your Python terminal?
```py
import ansys.geometry.core as pygeometry

print(pygeometry.__version__)
```

**Reply by t0xine** [reply]
Hi,
Thank you for the answer!
From pip freeze I can see all the sub-package of pyansys (ansys-geometry-core, ansys-mechanical-core...) and pyansys==2024.1.8
![image](https://github.com/ansys/pyansys/assets/170089887/3ea02ec8-d721-4d60-902a-de5aadfb7079)

**Reply by RobPasMue** [reply]
I believe your IDE (VSCode if I am not mistaken) is getting the wrong Python interpreter. Do you have your dependencies installed in a virtual environment? If so, you should make use of that specific interpreter. One thing you can try is:

1. Activate your virtual environment (on windows: ``.\.venv\Scripts\activate``
2. Run your script right after: ``python ansys.py``

**Comment by t0xine** [comment]
Here is what I have done (everything on the Command Prompt):

- Create a venv environment (py -m venv myenv)
- Activate it (myenv\Scripts\activate)
- Run ansys.py 
(Traceback (most recent call last):
  File "C:\Users\x\Documents\vs code\ansys.py", line 1, in <module>
    import ansys.geometry.core as pygeometry
  File "C:\Users\x\Documents\vs code\ansys.py", line 1, in <module>
    import ansys.geometry.core as pygeometry
ModuleNotFoundError: No module named 'ansys.geometry'; 'ansys' is not a package)

- Download pyansys on the venv (pip install pyansys)
- Check for the version (pyansys==2024.1.8)
- Run again (same output as previously)

**Reply by RobPasMue** [reply]
How are you running the script? Through your IDE or through the command line?

**Reply by t0xine** [reply]
I try both

**Reply by RobPasMue** [reply]
Make sure your IDE is pointing towards your virtual environment:

![image](https://github.com/ansys/pyansys/assets/37798125/5b466e80-905a-4610-89f1-9b99caab0c79)

In my case I am using the Python interpreter in my virtual environment.

Another option is to do the following...

1. Go to your ansys.py folder on the terminal
2. ``python -m venv .venv``
3. ``.venv\Scripts\activate``
4. ``python -m pip install pyansys``
5. ``python ansys.py``

**Comment by t0xine** [comment]
![image](https://github.com/ansys/pyansys/assets/170089887/9fe70bc4-26f4-4063-844f-af856116805e)
![image](https://github.com/ansys/pyansys/assets/170089887/b38545a9-a72b-4c1b-a954-08035d533dce)
![image](https://github.com/ansys/pyansys/assets/170089887/515fcf71-a6ba-476e-8ae6-63e8e6568209)
Numpy is working perfectly but not PyAnsys

**Reply by RobPasMue** [reply]
This is very weird, first time I see it...

Can you verify that ``.venv2\Lib\site-packages\ansys`` exists?

**Reply by t0xine** [reply]
C:\Users\lac\Documents\vs code\.venv2\Lib\site-packages\ansys\geometry\core
and all other sub-package

**Reply by RobPasMue** [reply, marked as answer]
Hmm...

I'm worried about the "vs code" folder. Spaces are typically a problem... Can we do as follows? You will just need to copy paste my commands on a terminal.

0. Open a new PowerShell terminal (fresh, not inside VSCode)
1. Create a folder in documents called pygeometry_test: ``mkdir C:\Users\lac\Documents\pygeometry_test``
2. Move to this folder: ``cd  C:\Users\lac\Documents\pygeometry_test``
3. Create a virtual environment: ``python -m venv .venv``
4. Activate the venv: ``.venv\Scripts\activate``
5. Install pyansys: ``python -m pip install pyansys``
6. Open a Python terminal: ``python``
7. Run the following:
```py
from ansys.geometry.core import __version__

print(__version__)
```

Please, follow each command step by step. If you encounter any problems in the way let me know

**Reply by t0xine** [reply]
![image](https://github.com/ansys/pyansys/assets/170089887/ad43e50f-8c07-45c4-b345-91aa062a0ad6)
![image](https://github.com/ansys/pyansys/assets/170089887/0d6e5d80-341b-4bc3-8ced-abaadd9567c1)

Oh great it's working!

**Reply by RobPasMue** [reply]
Cool! Like I said... I think the blank spaces are playing with your Python interpreter somehow. What you can try and do now is to open VSCode in this folder specifically. Let me know how it goes!

**Reply by t0xine** [reply]
It seems that everything is working properly. Thanks you a lot for your much needed help.
Best!

**Reply by RobPasMue** [reply]
Anytime! Glad I was able to help!
